cmake/package.cmake

```diff
@@ -0,0 +1,91 @@
+# This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+#
+# This file is free software; as a special exception the author gives
+# unlimited permission to copy and/or distribute it, with or without
+# modifications, as long as this notice is preserved.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY, to the extent permitted by law; without even the
+# implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+
+if("${SCRIPTS}" MATCHES "dynamic")
+  message(FATAL_ERROR "Packaging doesn't support dynamic scripts.")
+  return()
+endif()
+
+# Define install candidates for SQL files
+include(GNUInstallDirs)
+install(DIRECTORY "${CMAKE_SOURCE_DIR}/sql" DESTINATION "${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}")
+
+# Download the full database
+set(rev_database_version "${rev_version_major}${rev_version_minor}${rev_version_patch}")
+set(FULL_DATABASE_URL "https://github.com/TrinityCore/TrinityCore/releases/download/TDB335.${rev_database_version}/TDB_full_world_335.${rev_database_version}_${rev_database_date}.7z"
+    CACHE STRING
+    "The url to download the full database from"
+)
+message("* Package database URL   : ${FULL_DATABASE_URL}")
+get_filename_component(FULL_DATABASE_NAME "${FULL_DATABASE_URL}" NAME)
+string(REGEX REPLACE "\\.[^.]*$" "" FULL_DATABASE_NAMEWE ${FULL_DATABASE_NAME})
+if(NOT EXISTS "${CMAKE_CURRENT_BINARY_DIR}/${FULL_DATABASE_NAME}")
+  file(DOWNLOAD
+    "${FULL_DATABASE_URL}"
+    "${CMAKE_CURRENT_BINARY_DIR}/${FULL_DATABASE_NAME}"
+  )
+
+  execute_process(
+    COMMAND
+      ${CMAKE_COMMAND} -E tar xzf "${CMAKE_CURRENT_BINARY_DIR}/${FULL_DATABASE_NAME}"
+    WORKING_DIRECTORY
+      ${CMAKE_CURRENT_BINARY_DIR}
+  )
+endif()
+
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${FULL_DATABASE_NAMEWE}.sql" DESTINATION "${CONF_DIR}")
+
+if(NOT CPACK_GENERATOR)
+  set(CPACK_GENERATOR "DEB")
+endif()
+if(SERVERS AND TOOLS)
+  set(CPACK_PACKAGE_NAME "trinity")
+  set(CPACK_DEBIAN_PACKAGE_CONFLICTS "trinity-server, trinity-tools")
```

Comment:
You should make this depend on branch too so that both 3.3.5 and master can be installed side by side

Reply:
I think that would require renaming the binaries too, otherwise they would conflict

Reply:
Kind of want this fixed actually, binaries dont need to be renamed if you just install to a different directory (CMAKE_INSTALL_PREFIX or CPACK_PACKAGE_INSTALL_DIRECTORY)